### ISSUE TITLE
FilterResposeCurve fails in emscripten

### ISSUE THREAD

**Issue by baconpaul** [surge-synthesizer/surge#6733]
Over in https://github.com/surge-synthesizer/surge-rack/pull/702 @falkTX found that the response curve renderer would crash in wasm which is unaligned memory by default

I wonder if we forgot an alignas either there or in the subordinate analyzer?

he also helpfully told me how to run tests (copied below) in wasm so I can even try and replicate the crash

Wonder if the filter analyzer runs reliably in 32 bit windows? That would have the same problem but our use case there is so low anyway.

-------

Not too complex, https://emscripten.org/docs/getting_started/downloads.html details how to install.
I then just do:

source path/to/emsdk/emsdk_env.sh
cd /path/to/sst-filters
emcmake cmake -DSST_FILTERS_BUILD_TESTS=TRUE .
make
then make any dumb html code that loads the generate js file, like so:

<!doctype html>
<html lang="en-us">
  <head>
    <script async type="text/javascript" src="sst-filters-tests.js"></script>
  </head>
  <body>
  </body>
</html>
Then load up the html file on a browser, might need to be served through a web server for security considerations.
Easiest is just emrun tests/sst-filters-tests.html

**Comment by baconpaul** [comment]
yeah it's absolutely the case that we use a `std::vector<float>` which is not aligned at all in `set-filters plotFilterMagnitudeResponse`

probably breaks the filter analyzer on 32 bit windows also.

**Comment by baconpaul** [comment]
but we read that out 1 by 1 and do a set1_ps

it may be the delay buffers also

but basically this means i gotta actually try it on a non-aligned system and see where it blows out. This will stay in 1x

**Comment by baconpaul** [comment]
this bugs me enough that i want to look before 1.2 so moving it back into the 12 milestone

**Comment by mx** [comment]
If it's easy to change the vector, you can use my aligned allocator from here: https://github.com/surge-synthesizer/pffft/blob/surge/pffft.hpp

Probably move that out of pffft and into sst-cpputils.

**Comment by baconpaul** [comment]
the vector isn't the problem it turns out. but i'm not sure what is (or if there even is one).

appreciate it though!

**Comment by baconpaul** [comment]
Hmm ok in windows 32 bit this works so it really is only an emscripten problem.
I'm going to move this to currently unscheduled and rename it as a result.

**Comment by weepy** [comment]
would it be possible to run this in the browser?
